drivers/video/sunxi/sunxi_display.c

```diff
@@ -2368,14 +2371,20 @@ void *video_hw_init(void)
   lcd_init();
 
   uint16_t bug=3;
+  rc = run_command("load mmc 0:1 0x80000000 miyoo-boot.bmp", 0);
+
   while(bug--){
     uint16_t x, y;
-    uint32_t cnt=0;
-    uint16_t *p = (uint16_t*)logo;
-
-		if(miyoo_ver != 3){
+    if(rc == 0) {
+        cnt=13; //skip BMP header
+        p = (uint16_t*)0x80000000;
+    } else {
+        cnt=0;
```

Comment:
Do we need `else` here to reassign the same values? I thought that video_hw_init is one time run.

Reply:
there is a loop while(bug--) I don't know why, but there must be a reason in the past

Reply:
must be a reason while we loop 3 times, ok let's keep it that way

Reply:
I suspect that the some panels are not yet ready (long initialization time), and when it starts to draw the logo, after the first loop, it is incomplete.